Lib/email/feedparser.py

```diff
@@ -329,10 +606,11 @@ def _parsegen(self):
             # this onto the input stream until we've scanned past the
             # preamble.
             separator = '--' + boundary
-            def boundarymatch(line):
-                if not line.startswith(separator):
+            def boundarymatch(line, pos = 0, endpos = sys.maxsize):
```

Comment:
I'm not sure why we need endpos here? In my thought all of the `boundarymatch` is called without endpos. So all of the string endpos will be sys.maxsiz.